### PR TITLE
add patch for GROMACS 2024.1 to fix filesystem race in tests

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
CORE_CNT=16

File: easybuild/easyconfigs/g/GROMACS/GROMACS-2024.1-foss-2023b.eb [hunk 1]
```diff
@@ -72,8 +75,7 @@ exts_default_options = {
 
 exts_list = [
     ('gmxapi', '0.4.2', {
-        'preinstallopts': 'export CMAKE_ARGS="-Dgmxapi_ROOT=%(installdir)s ' +
-                          '-C %(installdir)s/share/cmake/gromacs_mpi/gromacs-hints_mpi.cmake" && ',
+        'preinstallopts': 'export CMAKE_ARGS="-Dgmxapi_ROOT=%(installdir)s -C %(installdir)s/share/cmake/gromacs_mpi/gromacs-hints_mpi.cmake" && ',
```
Comment (ocaisa) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2024.1-foss-2023b.eb:78]:
```suggestion
        'preinstallopts': 'export CMAKE_ARGS="-Dgmxapi_ROOT=%(installdir)s -C ' +
                                    '%(installdir)s/share/cmake/gromacs_mpi/gromacs-hints_mpi.cmake" && ',
```
Revert the change to pass CI

Comment (ocaisa) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-2024.1-foss-2023b.eb:78]:
```suggestion
        'preinstallopts': 'export CMAKE_ARGS="-Dgmxapi_ROOT=%(installdir)s ' +
                          '-C %(installdir)s/share/cmake/gromacs_mpi/gromacs-hints_mpi.cmake" && ',
```
Revert this change so CI can pass

**Review by ocaisa — APPROVED** [review]
Fixed a linting issue that doesn't affect the test builds

**Pull request merged**